### PR TITLE
Replace http with https whenever possible

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: guides/deployment/deployment.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Keep in mind that, if you by any chance forget to run the steps above, Phoenix w
 
 ```console
 $ PORT=4001 MIX_ENV=prod mix phx.server
-10:50:18.732 [info] Running MyAppWeb.Endpoint with Cowboy on http://example.com
+10:50:18.732 [info] Running MyAppWeb.Endpoint with Cowboy on https://example.com
```
Comment (mcrumm) [inline guides/deployment/deployment.md:60]: I do not think we should change this URL– the current value matches the default Endpoint configuration from the Phoenix installer.
```suggestion
10:50:18.732 [info] Running MyAppWeb.Endpoint with Cowboy on http://example.com
```

File: guides/deployment/deployment.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ To run Phoenix in production, we need to set the `PORT` and `MIX_ENV` environmen
 
 ```console
 $ PORT=4001 MIX_ENV=prod mix phx.server
-10:59:19.136 [info] Running MyAppWeb.Endpoint with Cowboy on http://example.com
+10:59:19.136 [info] Running MyAppWeb.Endpoint with Cowboy on https://example.com
```
Comment (mcrumm) [inline guides/deployment/deployment.md:72]: I do not think we should change this URL– the current value matches the default Endpoint configuration from the Phoenix installer.
```suggestion
10:59:19.136 [info] Running MyAppWeb.Endpoint with Cowboy on http://example.com
```

File: guides/deployment/deployment.md [hunk 3]
```diff
@@ -84,7 +84,7 @@ You can also run your application inside an interactive shell:
 
 ```console
 $ PORT=4001 MIX_ENV=prod iex -S mix phx.server
-10:59:19.136 [info] Running MyAppWeb.Endpoint with Cowboy on http://example.com
+10:59:19.136 [info] Running MyAppWeb.Endpoint with Cowboy on https://example.com
```
Comment (mcrumm) [inline guides/deployment/deployment.md:87]: I do not think we should change this URL– the current value matches the default Endpoint configuration from the Phoenix installer.
```suggestion
10:59:19.136 [info] Running MyAppWeb.Endpoint with Cowboy on http://example.com
```

File: lib/phoenix/router.ex [hunk 4]
```diff
@@ -121,10 +121,10 @@ defmodule Phoenix.Router do
       "/pages/hello?some=query"
 
       MyAppWeb.Router.Helpers.page_url(conn_or_endpoint, :show, "hello")
-      "http://example.com/pages/hello"
+      "https://example.com/pages/hello"
```
Comment (mcrumm) [inline lib/phoenix/router.ex:124]: I do not think we should change this URL– the current value matches the default Endpoint configuration from the Phoenix installer.
```suggestion
      "http://example.com/pages/hello"
```

File: lib/phoenix/router.ex [hunk 5]
```diff
@@ -121,10 +121,10 @@ defmodule Phoenix.Router do
...
 
       MyAppWeb.Router.Helpers.page_url(conn_or_endpoint, :show, "hello", some: "query")
-      "http://example.com/pages/hello?some=query"
+      "https://example.com/pages/hello?some=query"
```
Comment (mcrumm) [inline lib/phoenix/router.ex:127]: I do not think we should change this URL– the current value matches the default Endpoint configuration from the Phoenix installer.
```suggestion
      "http://example.com/pages/hello?some=query"
```

**Review by mcrumm — COMMENTED** [review]
Thanks @eksperimental! I left just a couple comments about URLs that I do not think we want to change.

**Comment by eksperimental** [comment]
> Thanks @eksperimental! I left just a couple comments about URLs that I do not think we want to change.

I agree with you @mcrumm 
Thank you.

**Comment by mcrumm** [comment]
Thanks! 💚

**Pull request merged**